backend/src/main/java/com/team28/sidedish/service/ProductService.java

```diff
@@ -1,7 +1,41 @@
 package com.team28.sidedish.service;
 
-import com.team28.sidedish.controller.dto.ProductListResponse;
+import com.team28.sidedish.exception.ProductNotFoundException;
+import com.team28.sidedish.repository.CategoryRepository;
+import com.team28.sidedish.repository.ProductRepository;
+import com.team28.sidedish.repository.entity.CategoryEntity;
+import com.team28.sidedish.repository.entity.ProductEntity;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
 
-public interface ProductService {
-    ProductListResponse findProductList(Long categoryId);
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class ProductService {
+
+    private final CategoryRepository categoryRepository;
+    private final ProductRepository productRepository;
+
+    public List<ProductEntity> findProducts(Long categoryId) {
+        Optional<CategoryEntity> categoryFindResult = categoryRepository.findById(categoryId);
+
+        if (categoryFindResult.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        CategoryEntity category = categoryFindResult.get();
+
+        return category.getProducts()
+                .stream()
+                .collect(Collectors.toUnmodifiableList());
+    }
```

Comment:
products을 리스트 컬렉션을 사용하면 스트림없이 가능할것 같네요
```suggestion
        return Collections.unmodifiableList(list)
    }
```

Reply:
처음에 Entity 를 만들 당시에는 Set 에 담아도 괜찮을거라 생각했는데, 
언급해주신대로 List 에 담는게 개발하기 훨씬 편하다는 생각을 많이 했습니다.
이 부분은 한번 개선해보겠습니다!
감사합니다!